src/App.vue

```diff
@@ -2,24 +2,28 @@
 import { ref } from "vue";
 import MCQQuiz from "@components/MCQ/MCQQuiz.vue";
 import { questions } from "@data/question-data.json";
+import StartPage from "./components/MCQ/StartPage.vue";
 const showQuiz = ref(false);
+const quizQuestions = ref(questions);
 
-const startQuiz = () => {
+const handleStartQuiz = (questionAmount: number) => {
+  quizQuestions.value = questions.slice(0, questionAmount);
   showQuiz.value = true;
 };
 </script>
 
 <template>
-  <MCQQuiz v-if="showQuiz" :questions="questions" />
-  <div v-else>
-    <h1>Welcome to VetsCloud Smart Quiz</h1>
-    <button class="start-button" @click="startQuiz">Start</button>
-  </div>
+  <MCQQuiz v-if="showQuiz" :questions="quizQuestions" />
+  <StartPage v-else :questions="questions" @start-quiz="handleStartQuiz" />
 </template>
 
```

Comment:
Would this be better handled with `xstate` or `vue dynamic-components`? (or even `vue-router` although I'm not sure about the urls)

Should the `StartPage` be responsible for the question filtering, simply passing the filtered question list straight to the Quiz?
Limit the `App` to managing which page to display - one way flow.

??? Something to think about.

Reply:
MCQ Quiz can possibly be rendered as a lead off from StartPage but I fear that might lead to coupling. Vue Router can also be used and could be a way to tackle this but then we might be introducing URLs as part of it. There is a way of using <component/> but instead of having a an if/else, it would still remain conditional. Will research more as I'm sure there's a better way of doing this. I might lean towards the router. Did you want me to do it as part of this task (since it's been approved)?

Reply:
No go ahead and merge. Just something to think about.

Considerations:
* who is responsible
* can we make it cleaner